src/Transport/Addressing/Partitioning/ReadOnlySettingsExtensions.cs

```diff
@@ -0,0 +1,22 @@
+﻿namespace NServiceBus.AzureServiceBus.Addressing.Partitioning
+{
+    using System.Collections.Generic;
+    using System.Configuration;
+    using System.Linq;
+    using Settings;
+    using Transport.AzureServiceBus;
+
+    public static class ReadOnlySettingsExtensions
+    {
+        public static List<RuntimeNamespaceInfo> GetPartitioningNamespaces(this ReadOnlySettings settings)
```

Comment:
Custom topology will still rely on injected `ReadOnlySettings` and if we still do that, I am not sure what's the benefit of hiding `NamespaceConfigurations`. Might just ship it as public for now and introduce a proper (breaking) API in the next major.